### ISSUE TITLE
Use notarytool on macOS

### ISSUE THREAD

**Issue by philrz** [brimdata/zui#2779]
Today we received the following notice via email from Apple: 

> We’re reaching out because you recently used the altool command-line utility to notarize your macOS software with Apple. As announced last year at [WWDC22](https://developer.apple.com/videos/play/wwdc2022/10109), if you’re still using altool with the Apple notary service, you should transition to the notarytool command-line utility as soon as possible. Notarizing software with altool was deprecated in Xcode 13, and the Apple notary service will no longer accept uploads from altool as of November 1, 2023. Existing notarized software will continue to function properly. 
>
> For information on notarizing your apps, read [TechNote TN3147: Migrating to the latest notarization tool](https://developer.apple.com/documentation/technotes/tn3147-migrating-to-the-latest-notarization-tool?changes=_3_3).

I did a quick search and it appears that the only current reference in the Zui repo that I can find to `altool` is in [CONTRIBUTING.md](https://github.com/brimdata/zui/blob/main/CONTRIBUTING.md) where we show a command line for checking the status of a notarization in progress. So if there's any truth to Apple's statement that we've been spotted using `altool` for actual notarization, it would imply we're making some second-order reference to it.

**Comment by philrz** [comment]
I did some digging and found this line:

https://github.com/brimdata/zui/blob/6022ebed5cacd35a64b449c975842a929d9db810/electron-builder.json#L17

That indicates we currently use [electron-builder-notarize](https://github.com/karaggeorge/electron-builder-notarize) to do our notarization after signing. Zui's tip of `main` has the dependency in `package.json` defined as:

https://github.com/brimdata/zui/blob/6022ebed5cacd35a64b449c975842a929d9db810/package.json#L103

Which has resolved since #2414 (almost a year ago) as:

https://github.com/brimdata/zui/blob/6022ebed5cacd35a64b449c975842a929d9db810/yarn.lock#L5991-L5993

It looks like `electron-builder-notarize` version `1.2.0` points at [electron/notarize](https://github.com/electron/notarize) version `^0.2.0`, which was tagged well before the changes in https://github.com/electron/notarize/pull/82 had `notarytool` being used for the first time. Also, [this code](https://github.com/karaggeorge/electron-builder-notarize/blob/f78f7644edb8a0554c2d5a99e9aad72f537f70ff/validate.js#L40-L50) in `electron-builder-notarize` version `1.5.0` indicates that it tries to call `electron-notarize` with a config to invoke `notarytool` if the needed config/binaries are present. So to summarize, this indicates that at one time our builds were definitely _not_ invoking `notarytool`, but for at least a year we've been pointing at tools that _can_ use `notarytool` and probably do.

---

Before I starting digging more into that, I spotted this comment in the link from the original Apple message:

> The submissions IDs returned by `notarytool` live in a different namespace than those returned by `altool`. You can’t check on the status of an `altool` submission with `notarytool`, or vice versa.

So, there's the `altool` command line we've had in our `CONTRIBUTING.md`:

```
xcrun altool --notarization-history 0 -u <user> -p <app-specific-password>
```

And there's the similar `notarytool` command line:

```
xcrun notarytool history --apple-id <user> --password <app-specific-password> --team-id <team-id>
```

This actually provides a handy black box testing approach for seeing which tool is being invoked, since I found I could start a Dev Build and then check both those tools and see a new ID/timestamp appear in only one of them. Using that approach revealed two interesting things:

1. Builds in https://github.com/brimdata/zui consistently end up causing entries to appear in the `notarytool` output, indicating `notarytool` was used for notarization. This lines up with my expectations after my findings logged in the first half of this comment.

2. Builds in https://github.com/brimdata/zui-insiders end up causing entries to appear in the `altool` output, indicating `altool` was used for notarization. Given that Zui Insiders derives its build config all from Zui, this does _not_ make sense to me. In reviewing this findings with @jameskerr the current theory is that the difference is related to cached dependencies somehow still causing `electron-builder-notarize` version `1.2.0` to be used when Zui Insiders is being built.

**Comment by philrz** [comment]
Update: Now that it's been a couple weeks since this change, here's verification that there's not been any additional notarizations with `altool`. So I think this one is officially behind us.

```
$ xvrun altool --notarization-history 0 -u "phil@brimdata.io" -p xxxx-xxxx-xxxx-xxxx
Notarization History - page 0

Date                      RequestUUID                          Status   Status Code Status Message   
------------------------- ------------------------------------ -------- ----------- ---------------- 
2023-06-15 18:52:41 +0000 f8c37dbf-b84e-4c39-bbf7-5e015cb86bce success  0           Package Approved 
2023-06-15 15:14:31 +0000 1ab1cac5-776f-41b8-86b3-76b4ebb4145e success  0           Package Approved 
2023-06-15 02:23:12 +0000 352195e1-1116-4dbe-aa76-4821b3c1ac3a success  0           Package Approved 
2023-06-15 00:00:46 +0000 a7497fd5-c240-427d-b3b6-66fb97867a99 success  0           Package Approved 
2023-06-14 23:24:12 +0000 0be3c4dd-6805-4f56-bf53-9080a18f807a success  0           Package Approved 
...
```
